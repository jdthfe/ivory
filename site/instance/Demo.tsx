import React from 'react';
import { getInstance } from '../_util';

interface DemoProps {
    item: ComponentIndex;
}
const Demo = (props: DemoProps) => {
    const { item } = props;
    const Demo = getInstance(item) || null;
    return <main className="instance-demo">{Demo ? <Demo /> : null}</main>;
};

export default Demo;
