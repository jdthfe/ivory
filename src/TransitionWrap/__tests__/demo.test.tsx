import webDemoTest from '@tests/shared/demoTest';
webDemoTest('TransitionWrap');

import React from 'react';
import Demo from '../demo';
import { render, fireEvent, waitForElement, waitForElementToBeRemoved } from 'react-testing-library';

describe('TransitionWrapDemo', () => {
    it('unmontOnExit', async () => {
        const { baseElement, getAllByRole } = render(<Demo />);
        const btn = getAllByRole('button')[0];
        // https://testing-library.com/docs/dom-testing-library/cheatsheet#queries
        fireEvent.click(btn);

        // https://testing-library.com/docs/dom-testing-library/api-async
        await waitForElement(() => baseElement.querySelector(`.hello-entry`));
        await waitForElement(() => baseElement.querySelector(`.hello-entry-active`));
        await waitForElement(() => baseElement.querySelector(`.hello-entry-done`));

        expect(baseElement).toMatchSnapshot('visible=true');
        fireEvent.click(btn);

        await waitForElement(() => baseElement.querySelector(`.hello-exit`));
        await waitForElement(() => baseElement.querySelector(`.hello-exit-active`));
        await waitForElementToBeRemoved(() => baseElement.querySelector(`.hello-exit-active`));
        // await waitForElement(() => baseElement.querySelector(`.hello-exit-done`));
        expect(baseElement).toMatchSnapshot('visible=false');
    });

    it('keepOnExit', async () => {
        const { baseElement, getAllByRole } = render(<Demo />);
        const btn = getAllByRole('button')[1];

        await waitForElement(() => baseElement.querySelector(`.world-exit-done`));
        fireEvent.click(btn);

        await waitForElement(() => baseElement.querySelector(`.world-entry`));
        await waitForElement(() => baseElement.querySelector(`.world-entry-active`));
        await waitForElement(() => baseElement.querySelector(`.world-entry-done`));

        expect(baseElement).toMatchSnapshot('visible=true');
        fireEvent.click(btn);

        await waitForElement(() => baseElement.querySelector(`.world-exit`));
        await waitForElement(() => baseElement.querySelector(`.world-exit-active`));
        await waitForElement(() => baseElement.querySelector(`.world-exit-done`));
        expect(baseElement).toMatchSnapshot('visible=false');
    });
});
