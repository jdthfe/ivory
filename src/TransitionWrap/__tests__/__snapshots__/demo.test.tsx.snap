// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransitionWrapDemo keepOnExit: visible=false 1`] = `
<body>
  <div>
    <div
      class="eui-wingblank TransitionWrap eui-wingblank-m"
    >
      <button
        class="eui-button eui-button-l eui-button-customize eui-button-ghost"
        role="button"
      >
        keepOnExit = false
      </button>
      <div
        class="eui-whitespace eui-whitespace-m"
      />
      <button
        class="eui-button eui-button-l eui-button-customize eui-button-ghost"
        role="button"
      >
        keepOnExit = true
      </button>
      <h1
        class="world world-exit-done"
      >
        world
      </h1>
    </div>
  </div>
</body>
`;

exports[`TransitionWrapDemo keepOnExit: visible=true 1`] = `
<body>
  <div>
    <div
      class="eui-wingblank TransitionWrap eui-wingblank-m"
    >
      <button
        class="eui-button eui-button-l eui-button-customize eui-button-ghost"
        role="button"
      >
        keepOnExit = false
      </button>
      <div
        class="eui-whitespace eui-whitespace-m"
      />
      <button
        class="eui-button eui-button-l eui-button-customize eui-button-ghost"
        role="button"
      >
        keepOnExit = true
      </button>
      <h1
        class="world world-entry-done"
      >
        world
      </h1>
    </div>
  </div>
</body>
`;

exports[`TransitionWrapDemo unmontOnExit: visible=false 1`] = `
<body>
  <div>
    <div
      class="eui-wingblank TransitionWrap eui-wingblank-m"
    >
      <button
        class="eui-button eui-button-l eui-button-customize eui-button-ghost"
        role="button"
      >
        keepOnExit = false
      </button>
      <div
        class="eui-whitespace eui-whitespace-m"
      />
      <button
        class="eui-button eui-button-l eui-button-customize eui-button-ghost"
        role="button"
      >
        keepOnExit = true
      </button>
      <h1
        class="world world-exit-done"
      >
        world
      </h1>
    </div>
  </div>
</body>
`;

exports[`TransitionWrapDemo unmontOnExit: visible=true 1`] = `
<body>
  <div>
    <div
      class="eui-wingblank TransitionWrap eui-wingblank-m"
    >
      <button
        class="eui-button eui-button-l eui-button-customize eui-button-ghost"
        role="button"
      >
        keepOnExit = false
      </button>
      <div
        class="eui-whitespace eui-whitespace-m"
      />
      <button
        class="eui-button eui-button-l eui-button-customize eui-button-ghost"
        role="button"
      >
        keepOnExit = true
      </button>
      <h1
        class="hello hello-entry-done"
      >
        hello
      </h1>
      <h1
        class="world world-exit-done"
      >
        world
      </h1>
    </div>
  </div>
</body>
`;

exports[`renders TransitionWrap correctly 1`] = `
<div>
  <div
    class="eui-wingblank TransitionWrap eui-wingblank-m"
  >
    <button
      class="eui-button eui-button-l eui-button-customize eui-button-ghost"
      role="button"
    >
      keepOnExit = false
    </button>
    <div
      class="eui-whitespace eui-whitespace-m"
    />
    <button
      class="eui-button eui-button-l eui-button-customize eui-button-ghost"
      role="button"
    >
      keepOnExit = true
    </button>
    <h1
      class="world world-exit-done"
    >
      world
    </h1>
  </div>
</div>
`;
